### PR TITLE
Refactor store; remove ducks; standardize syntax

### PR DESCRIPTION
**Pull request by louh**
At the beginning of this project I was giving [Ducks](https://github.com/erikras/ducks-modular-redux) a try -- it's an alternative strategy to getting around some of the [boilerplate](http://redux.js.org/docs/recipes/ReducingBoilerplate.html) that idiomatic Redux asks you to write.

Dan Abramov (creator of Redux) has said he does not necessarily approve of Ducks because it can enforce a [1:1 relationship between actions and reducers](https://twitter.com/dan_abramov/status/738405796770353152) and I've now hit on a scenario (in another branch) where this becomes a problem. For instance, say we issue a `RESET_EVERYTHING` action. Different reducers should be able to listen for this action so that each of them can clear their state. But in the Ducks format, there's no idiomatic way to store shared actions. We could presumably create a shared action module to import from.

However, Ducks was not used as a standard throughout, so rather than convert everything to Ducks and then work around its issues, I've decided to drop it entirely and go back to plain ol' Redux. Ideally this is better for maintainability, if a little more verbose, in that everything one needs to maintain the Redux store can be gleaned from the regular Redux documentation. This might not hold 100% true moving forward, but I'm comfortable making this reset for the time being.

This PR removes Ducks, and standardizes some of the Redux syntax to be consistent throughout.

### CODE REVIEW

**Comment by mkong0216** [comment]
Question. What's the advantage/point of initializing a variable to a string of the same value and then exporting it? Like say `export const ADD_ROUTE_WAYPOINT = 'ADD_ROUTE_WAYPOINT'`

**Comment by louh** [comment]
That's a great question. The Redux docs describes some of the thought process behind it here: http://redux.js.org/docs/recipes/ReducingBoilerplate.html#actions

I think using string values as-is has some validity because it is simple and clean, and you could easily do a search-and-replace throughout the entire project if you need to rename a string. But, after some time working with Redux on large projects, I find that the most valuable reasons to export constants are these:

> Sometimes you want to see all existing actions before working on a new feature

If we didn't export them out of a central place, it eventually becomes really hard to know what all the actions are. (I guess you could skim through all the reducers.)

> If you make a typo when importing an action constant, you will get `undefined`. 

Basically, built-in error checking. If the constant doesn't exist, you could catch the error during a test. String typos "fall through", instead of triggering the bug at runtime.

So why is the value a string of the variable name itself? The thing is, only the variable name is important. The string itself can be anything. The catch is that the strings should be unique, otherwise you could have one action accidentally equal another one, and that might be confusing or have unexpected effects. One useful side effect of declaring all the constant variables in one place is that all the variable names must be unique or the tests will fail. So if the string equals the variable name, it's basically guaranteed to be a unique string in the project.
